### PR TITLE
docs: add a Why section to the readme

### PR DESCRIPTION
**Pull request by mikeshng**
I am trying to show maestro as a possible solution for another project but the README is missing some motivations on why this project is needed. This PR is to address that missing overview.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,12 +1,40 @@
...
 
 ## Architecture
+
 Taking MQTT as an example:
+
 ![Maestro Architecture](./arch.png)
 
+## Why
+
+Maestro was created to address several critical challenges associated with managing Kubernetes clusters at scale.
```
Comment (markturansky) [inline README.md:15]: This paragraph is good in that we should see performance and scalability improvements by using Maestro, as you note.

Another motivation (I think, perhaps, primary motivation) was to improve the security architecture.

By using Maestro, we do not need kubeconfigs.  Our Cluster Service (CS) deployment would no longer need kubeconfigs or direct access to customer clusters. CS just uses the Maestro API. 

Only the Maestro consumer in a management cluster needs kubeconfigs, thus improving security.

Comment (mikeshng) [reply README.md:15]: Thanks Mark for the review!

Could you please help me word it in a way that's more generic and not mention CS, SD, MC, HCP etc?
I am trying to show maestro as a possible solution for another project and it's not related to SD. Maybe closer to self managed?

Comment (mikeshng) [reply README.md:15]: Added a generic sentence about not needing kubeconfig. Maybe it's a bit vague but PTAL. Thanks.

**Comment by clyang82** [comment]
@mikeshng Thanks for your PR. Maybe add a description for the purpose of this PR. something like show maestro as a possible solution for another project.

**Comment by qiujian16** [comment]
lgtm

**Comment by clyang82** [comment]
/lgtm

leave it to @markturansky for another look.

**Review by clyang82 — APPROVED** [review]

**Pull request merged**